### PR TITLE
fix: Remove Node 12 and Node 17 support

### LINKED ISSUE #8081: Remove Node 12 and Node 17 support
### New Feature / Enhancement Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server/blob/master/SECURITY.md).
- [x] I am not just asking a [question](https://github.com/parse-community/.github/blob/master/SUPPORT.md).
- [x] I have searched through [existing issues](https://github.com/parse-community/parse-server/issues?q=is%3Aissue).

### Current Limitation
Node 12 and Node 17 are out of LTS, but its still in the CI and node engine

### Feature / Enhancement Description
Remove Node 12 and Node 17 support

Removing Parse Server compatibility would be a breaking change, so this can only be done in Parse Server 6.

### Example Use Case
n/a

### Alternatives / Workarounds
n/a

### PR DESCRIPTION
**Pull request by dblythy**
### New Pull Request Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server/blob/master/SECURITY.md).
- [x] I am creating this PR in reference to an [issue](https://github.com/parse-community/parse-server/issues?q=is%3Aissue).

### Issue Description

Related issue: #8081

### Approach

Removes Node 12 and Node 17 support

### TODOs before merging

- [x] Add tests
- [x] A changelog entry is created automatically using the pull request title (do not manually add a changelog entry)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -133,7 +133,7 @@
     "madge:circular": "node_modules/.bin/madge ./src --circular"
   },
   "engines": {
-    "node": ">=12.22.10 <19"
+    "node": ">=14.15.0 <19"
```
Comment (mtrezza) [inline package.json]: We don't plan on changing the node engine version after this until Dec 2023. Let's use the latest Node 14 version (14.21.0) to lower the risk of not being able to upgrade dependencies if they require a higher node version. This should then also be changed in the CI workflow where the Node 14 version to test is set.

**Review by mtrezza — CHANGES_REQUESTED** [review]

**Comment by mtrezza** [comment]
I have changed the issue and this PR description to include removal of Node 17. Node 17 also reached EOL in June 2022. Could you also remove Node 17 in this PR?

File: package.json (hunk 1 above)
Comment (dblythy) [reply package.json]: How should this look to exclude node 17?

**Comment by dblythy** [comment]
Sure. I also noticed that there is no `Node 18` step of the CI, which looks like it was removed in #8195. Is this intended?

**Comment by mtrezza** [comment]
Yes, because Node 18 is what the CI uses for all other tests by default, so there's no need to explicitly add a Node 18 environment.

File: package.json (hunk 1 above)
Comment (mtrezza) [reply package.json]: `>=14.21.0 <17 || >=18 <19`

Comment (dblythy) [reply package.json]: Thanks!!

**Review by mtrezza — APPROVED** [review]
Looks good!

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [6.0.0-alpha.1](https://github.com/parse-community/parse-server/releases/tag/6.0.0-alpha.1)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [6.0.0-beta.1](https://github.com/parse-community/parse-server/releases/tag/6.0.0-beta.1)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [6.0.0](https://github.com/parse-community/parse-server/releases/tag/6.0.0)

**Pull request merged**